### ISSUE TITLE
Is it possible to train it in a non-distributed way?

### ISSUE THREAD

**Issue by boli2016** [wenwenyu/PICK-pytorch#29]
Hi wenwenyu,

I am really interested in this work. Is it possible for us to train it in a non-distributed way?

Thanks a lot!

**Comment by wenwenyu** [comment]
Hi, 

Thanks for your interest in our work. It can be run in a non-distributed way but needs to modified code. I will support this training mode in this repo as soon as possible. I have lots of things to do, please stay tuned and wait patiently.

**Comment by tengerye** [comment]
@wenwenyu Hi, I have modified the code to make it run in a non-distributed way. I have also added an example as how to train it on a big dataset. I will push it this week.

**Comment by wenwenyu** [comment]
@tengerye Hi, I'm very glad that you have finished it. This is also an awesome work about adding a training example on big datasets. Pull requests are absolutely welcome.

**Comment by boli2016** [comment]
Thank you so much @wenwenyu @tengerye !

**Comment by tengerye** [comment]
@boli2016 @wenwenyu Hi, apologize. This [pull](https://github.com/wenwenyu/PICK-pytorch/pull/30) does not include the "non-distributed" way. There are two options here now:

1. We add "non-distributed" code and leave it different with "distributed" code;

2. We make python control the distribution instead of `torch.distributed.launch`. In other words, "distributed" codes works by invoking "non-distributed" codes. I prefer this way.

Dear @wenwenyu , I just send you an email including more details, would you please have a look at your convenience please?

**Comment by wenwenyu** [comment]
@tengerye Hi, I've received your email and I will reply you at my convince time.

**Comment by cloudfool** [comment]
@tengerye Hi, Could you PLZ send me the  "non-distributed" training code? TKS weihao1991@126.com

**Comment by tengerye** [comment]
@cloudfool Hi, sincerely sorry for the delay. I am quite busy recently. But we are working on this already. My plan is to continue to work on it and try to release it during the National Day.

**Comment by cloudfool** [comment]
TKS，keep tuned

**Comment by tengerye** [comment]
@cloudfool @boli2016 I have finished the non-distributed training part. Would you please take a shot?

**Comment by sanku015** [comment]
@tengerye could you share the codes?

**Comment by cloudfool** [comment]
@tengerye Thanks a lot
